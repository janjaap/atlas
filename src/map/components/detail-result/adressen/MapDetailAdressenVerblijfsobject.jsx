import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultItem from '../MapDetailResultItem';
import MapDetailResultStatusItem from '../MapDetailResultStatusItem';
import MapDetailResultWrapper from '../MapDetailResultWrapper';
import MapDetailAdressenVerblijfsobjectGebruiksdoelenItem from './MapDetailAdressenVerblijfsobjectGebruiksdoelenItem';

const statusToCssModifier = {
  18: 'alert'
};

const MapDetailAdressenVerblijfsobject = ({
  panoUrl, verblijfsobject, onMaximize, onPanoPreviewClick
}) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={verblijfsobject.label}
    title="Adres"
  >
    <ul className="map-detail-result__list">
      { verblijfsobject.gebruiksdoelen && <MapDetailAdressenVerblijfsobjectGebruiksdoelenItem
        gebruiksdoelen={verblijfsobject.gebruiksdoelen}
      /> }
      <MapDetailResultItem
        label="Oppervlakte"
        value={verblijfsobject.size ? `${verblijfsobject.size} mÂ²` : 'onbekend'}
      />
      <MapDetailResultItem
        label="Type woonobject"
        value={verblijfsobject.type}
      />
      <MapDetailResultItem
        label="Eigendomsverhouding"
        value={verblijfsobject.eigendomsverhouding}
      />
      <MapDetailResultStatusItem
        label="Indicatie hoofdadres"
        value={verblijfsobject.isNevenadres ? 'Nee' : 'Ja'}
        status={verblijfsobject.isNevenadres ? 'info' : ''}
      />
      <MapDetailResultStatusItem
        label="Status"
        value={verblijfsobject.status.description}
        status={statusToCssModifier[verblijfsobject.status.code]}
      />
    </ul>
  </MapDetailResultWrapper>
);

MapDetailAdressenVerblijfsobject.propTypes = {
  verblijfsobject: PropTypes.shape({
    eigendomsverhouding: PropTypes.string,
    gebruiksdoelen: PropTypes.array,
    label: PropTypes.string,
    size: PropTypes.number,
    type: PropTypes.string,
    status: PropTypes.shape({
      description: PropTypes.string,
      code: PropTypes.string
    })
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailAdressenVerblijfsobject;
