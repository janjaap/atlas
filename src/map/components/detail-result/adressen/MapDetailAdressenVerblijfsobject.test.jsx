import React from 'react';
import { shallow } from 'enzyme';

import MapDetailAdressenVerblijfsobject from './MapDetailAdressenVerblijfsobject';

describe('MapDetailAdressenVerblijfsobject', () => {
  it('should render everything', () => {
    const verblijfsobject = {
      eigendomsverhouding: 'Verblijfsobject eigendomsverhouding',
      gebruiksdoelen: [{
        code: '01',
        description: 'Gebruiksdoel description 1'
      }, {
        code: '0400',
        description: 'Gebruiksdoel description 2',
        descriptionPlus: 'Gebruiksdoel description plus'
      }],
      label: 'Verblijfsobject label',
      size: 15,
      type: 'Verblijfsobject type',
      status: {
        description: 'Status omschrijving',
        code: '15'
      },
      hoofdadres: {
        hoofdadres: true
      }
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailAdressenVerblijfsobject
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        verblijfsobject={verblijfsobject}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render unknown with zero value for size', () => {
    const verblijfsobject = {
      eigendomsverhouding: 'Verblijfsobject eigendomsverhouding',
      gebruiksdoelen: [{
        code: '01',
        description: 'Gebruiksdoel description 1'
      }, {
        code: '0400',
        description: 'Gebruiksdoel description 2',
        descriptionPlus: 'Gebruiksdoel description plus'
      }],
      label: 'Verblijfsobject label',
      size: 0,
      type: 'Verblijfsobject type',
      status: {
        description: 'Status omschrijving',
        code: '15'
      },
      hoofdadres: {
        hoofdadres: true
      }
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailAdressenVerblijfsobject
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        verblijfsobject={verblijfsobject}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render unknown without size', () => {
    const verblijfsobject = {
      eigendomsverhouding: 'Verblijfsobject eigendomsverhouding',
      gebruiksdoelen: [{
        code: '01',
        description: 'Gebruiksdoel description 1'
      }, {
        code: '0400',
        description: 'Gebruiksdoel description 2',
        descriptionPlus: 'Gebruiksdoel description plus'
      }],
      label: 'Verblijfsobject label',
      type: 'Verblijfsobject type',
      status: {
        description: 'Status omschrijving',
        code: '15'
      },
      hoofdadres: {
        hoofdadres: true
      }
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailAdressenVerblijfsobject
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        verblijfsobject={verblijfsobject}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render `Nee` if hoofdadres is `false` and show a blue bullet', () => {
    const verblijfsobject = {
      eigendomsverhouding: 'Verblijfsobject eigendomsverhouding',
      gebruiksdoelen: [{
        code: '01',
        description: 'Gebruiksdoel description 1'
      }, {
        code: '0400',
        description: 'Gebruiksdoel description 2',
        descriptionPlus: 'Gebruiksdoel description plus'
      }],
      label: 'Verblijfsobject label',
      type: 'Verblijfsobject type',
      status: {
        description: 'Status omschrijving',
        code: '15'
      },
      hoofdadres: {
        hoofdadres: false
      }
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailAdressenVerblijfsobject
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        verblijfsobject={verblijfsobject}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
