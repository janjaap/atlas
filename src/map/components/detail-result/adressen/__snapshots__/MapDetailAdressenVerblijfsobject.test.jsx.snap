// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailAdressenVerblijfsobject should render \`Nee\` if hoofdadres is \`false\` and show a blue bullet 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Verblijfsobject label"
  title="Adres"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailAdressenVerblijfsobjectGebruiksdoelenItem
      gebruiksdoelen={
        Array [
          Object {
            "code": "01",
            "description": "Gebruiksdoel description 1",
          },
          Object {
            "code": "0400",
            "description": "Gebruiksdoel description 2",
            "descriptionPlus": "Gebruiksdoel description plus",
          },
        ]
      }
    />
    <MapDetailResultItem
      label="Oppervlakte"
      value="onbekend"
    />
    <MapDetailResultItem
      label="Type woonobject"
      value="Verblijfsobject type"
    />
    <MapDetailResultItem
      label="Eigendomsverhouding"
      value="Verblijfsobject eigendomsverhouding"
    />
    <MapDetailResultStatusItem
      label="Indicatie hoofdadres"
      status=""
      value="Ja"
    />
    <MapDetailResultStatusItem
      label="Status"
      status=""
      value="Status omschrijving"
    />
  </ul>
</MapDetailResultWrapper>
`;

exports[`MapDetailAdressenVerblijfsobject should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Verblijfsobject label"
  title="Adres"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailAdressenVerblijfsobjectGebruiksdoelenItem
      gebruiksdoelen={
        Array [
          Object {
            "code": "01",
            "description": "Gebruiksdoel description 1",
          },
          Object {
            "code": "0400",
            "description": "Gebruiksdoel description 2",
            "descriptionPlus": "Gebruiksdoel description plus",
          },
        ]
      }
    />
    <MapDetailResultItem
      label="Oppervlakte"
      value="15 mÂ²"
    />
    <MapDetailResultItem
      label="Type woonobject"
      value="Verblijfsobject type"
    />
    <MapDetailResultItem
      label="Eigendomsverhouding"
      value="Verblijfsobject eigendomsverhouding"
    />
    <MapDetailResultStatusItem
      label="Indicatie hoofdadres"
      status=""
      value="Ja"
    />
    <MapDetailResultStatusItem
      label="Status"
      status=""
      value="Status omschrijving"
    />
  </ul>
</MapDetailResultWrapper>
`;

exports[`MapDetailAdressenVerblijfsobject should render unknown with zero value for size 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Verblijfsobject label"
  title="Adres"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailAdressenVerblijfsobjectGebruiksdoelenItem
      gebruiksdoelen={
        Array [
          Object {
            "code": "01",
            "description": "Gebruiksdoel description 1",
          },
          Object {
            "code": "0400",
            "description": "Gebruiksdoel description 2",
            "descriptionPlus": "Gebruiksdoel description plus",
          },
        ]
      }
    />
    <MapDetailResultItem
      label="Oppervlakte"
      value="onbekend"
    />
    <MapDetailResultItem
      label="Type woonobject"
      value="Verblijfsobject type"
    />
    <MapDetailResultItem
      label="Eigendomsverhouding"
      value="Verblijfsobject eigendomsverhouding"
    />
    <MapDetailResultStatusItem
      label="Indicatie hoofdadres"
      status=""
      value="Ja"
    />
    <MapDetailResultStatusItem
      label="Status"
      status=""
      value="Status omschrijving"
    />
  </ul>
</MapDetailResultWrapper>
`;

exports[`MapDetailAdressenVerblijfsobject should render unknown without size 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Verblijfsobject label"
  title="Adres"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailAdressenVerblijfsobjectGebruiksdoelenItem
      gebruiksdoelen={
        Array [
          Object {
            "code": "01",
            "description": "Gebruiksdoel description 1",
          },
          Object {
            "code": "0400",
            "description": "Gebruiksdoel description 2",
            "descriptionPlus": "Gebruiksdoel description plus",
          },
        ]
      }
    />
    <MapDetailResultItem
      label="Oppervlakte"
      value="onbekend"
    />
    <MapDetailResultItem
      label="Type woonobject"
      value="Verblijfsobject type"
    />
    <MapDetailResultItem
      label="Eigendomsverhouding"
      value="Verblijfsobject eigendomsverhouding"
    />
    <MapDetailResultStatusItem
      label="Indicatie hoofdadres"
      status=""
      value="Ja"
    />
    <MapDetailResultStatusItem
      label="Status"
      status=""
      value="Status omschrijving"
    />
  </ul>
</MapDetailResultWrapper>
`;
